### PR TITLE
ZCS-860 : Sieve: addheader/replaceheader must not add a header withou…

### PR DESCRIPTION
**Pull request by chinmay15**
Problem : header name used as space or starting with space should not be allowed.

Fix : left trimmed header name before validation.

Testing done : added unit tests for
1) addheader
    a) header name as single space should fail
    b) header name as multiple spaces should fail
    c) header name starting with space(s) should fail
2) replaceheader
    a) new header name as single space should fail
    b) new header name as multiple spaces should fail
    c) new header name starting with space(s) should fail
    d) header name mentioned as key should fail if same as a, b and c
3) deleteheader
    a) header name as single space should fail
    b) header name as multiple spaces should fail
    c) header name starting with space(s) should fail

Testing to be done by QA : verify all mentioned scenarios and do sanity around the same.

### CODE REVIEW

File: store/src/java/com/zimbra/cs/filter/jsieve/AddHeader.java [hunk 1]
```diff
@@ -154,6 +155,12 @@ protected void validateArguments(Arguments arguments, SieveContext context)
         }
 
         if (!StringUtil.isNullOrEmpty(headerName)) {
+            String tempHeaderName = headerName;
```
Comment (shrikant-prasad) [inline store/src/java/com/zimbra/cs/filter/jsieve/AddHeader.java]: should move this code to static method in FilterUtil class and call that method in AddHeader and EditHeaderExtension.

**Comment by chinmay15** [comment]
@shrikant-prasad done

File: store/src/java/com/zimbra/cs/filter/FilterUtil.java [hunk 2]
```diff
@@ -1029,5 +1029,14 @@ private static void validateVariableIndex(String srcStr) throws SyntaxException
...
        }
     }
-}
 
+    /**
+     * returns String by removing initial spaces
+     * @param text String to remove initial spaces
+     * @return
+     */
+    public static String removeInitialSpaces(String text) {
```
Comment (rohan-ambasta) [inline store/src/java/com/zimbra/cs/filter/FilterUtil.java]: May be you can use StringUtils.stripStart() instead of this method

**Comment by chinmay15** [comment]
@rohan-ambasta done

File: store/src/java/com/zimbra/cs/filter/jsieve/EditHeaderExtension.java [hunk 3]
```diff
@@ -299,11 +300,15 @@ public void setupEditHeaderData(Arguments arguments, AbstractCommand ac) throws
                         arg = itr.next();
                         if (arg instanceof StringListArgument) {
                             StringListArgument sla = (StringListArgument) arg;
-                            String newName = sla.getList().get(0);
-                            if (StringUtil.isNullOrEmpty(newName)) {
+                            String origNewName = sla.getList().get(0);
```
Comment (Prashantsurana) [inline store/src/java/com/zimbra/cs/filter/jsieve/EditHeaderExtension.java]: are  you sure to get index 0 from the list? Potential array index out of bound case here. 
in the next line there is check for empty or null string, it indicates there is possibility of empty list. 
Can you please confirm ?

Comment (chinmay15) [inline store/src/java/com/zimbra/cs/filter/jsieve/EditHeaderExtension.java]: Yes, that's correct. We are checking 1st if the arg is instance of  object of StringListArgument. StringListArguement is created when there is at least 1 String object in the list. So there is no scope for index out of bound.

Comment (Prashantsurana) [inline store/src/java/com/zimbra/cs/filter/jsieve/EditHeaderExtension.java]: Approving based on above comment.

**Comment by chinmay15** [comment]
merged in feature/sieve

**Comment by gordyt** [comment]
@chinmay15 When you get a chance, please do a clean rebase of the feature/sieve branch onto develop.  Please let me know if you have any questions.

**Comment by chinmay15** [comment]
@gordyt sure

**Pull request closed without merging**